boot/flags.go

```diff
@@ -161,27 +161,35 @@ func InitramfsActiveBootFlags(mode string, rootfsDir string) ([]string, error) {
 	case ModeInstall:
 		// boot flags always come from the bootenv of the recovery bootloader
 		// in install mode
+		return readBootFlagsFromRecoveryBootloader()
 
-		opts := &bootloader.Options{
-			Role: bootloader.RoleRecovery,
-		}
-		bl, err := bootloader.Find(InitramfsUbuntuSeedDir, opts)
-		if err != nil {
-			return nil, err
-		}
-
-		m, err := bl.GetBootVars("snapd_boot_flags")
-		if err != nil {
-			return nil, err
-		}
-
-		return splitBootFlagString(m["snapd_boot_flags"]), nil
+	case ModeFactoryReset:
+		// Reuse the code from ModeInstall as we have a lot of
+		// identical conditions.
+		return readBootFlagsFromRecoveryBootloader()
```

Comment:
Only if you like it better (if you plan to make the two cases different in the future, then definitely don't bother):
```go
	case ModeFactoryReset:
		// Reuse the code from ModeInstall as we have a lot of
		// identical conditions.
		fallthrough
	case ModeInstall:
		// boot flags always come from the bootenv of the recovery bootloader
		// in install mode
		return readBootFlagsFromRecoveryBootloader()
```

Reply:
(you could even specify both in a single `case` line, but then you'd have to tweak the comments to make the situation clear.)

Reply:
Ah really nice! I'll use the fall-through case to keep the comments seperate. But I really like the syntax of multiple values in a 'case'. Thanks for suggesting this!